cliquet/cache/memory.py

```diff
@@ -25,20 +26,24 @@ def flush(self):
         self._store = {}
 
     def ttl(self, key):
+        key = self.prefix + key
         ttl = self._ttl.get(key)
         if ttl is not None:
             return (ttl - utils.msec_time()) / 1000.0
         return -1
 
     def expire(self, key, ttl):
+        key = self.prefix + key
         self._ttl[key] = utils.msec_time() + int(ttl * 1000.0)
 
     def set(self, key, value, ttl=None):
-        self._store[key] = value
         if ttl is not None:
             self.expire(key, ttl)
+        key = self.prefix + key
```

Comment:
@Natim We can't do that because in `def expire()` we are again modifying the key. If we do it before calling the `self.expire()` then `key` will be modified to `prefix_prefix_key`